### PR TITLE
Test/ KLBP-104: Step required documents updates.

### PR DESCRIPTION
**Pull request by iuliudascalu**

### CODE REVIEW

**Review by ioan-iacob-netea — APPROVED** [review]

File: src/components/Step1/Step1Form.tsx [hunk 1]
```diff
@@ -14,13 +14,17 @@ const Step1Form: FC = () => {
...
   } = useStep1FormHook()
 
   return (
     <form className="row" onSubmit={handleSubmit(onSubmit)}>
       {/* TODO: Send details from the hook as props when continuing the flow without company information */}
       {requireBasicCompanyDetails && <BasicCompanyDetails />}
-      <UploadFiles {...{ dispatch, firstFile, onChange, secondFile }} />
+      <UploadFiles
+        {...{ dispatch, firstFile, secondFile, thirdFile, forthFile, onChange }}
+      />
```
Comment (andreir-suciu) [inline src/components/Step1/Step1Form.tsx:27]: Please try the following simplified update.
It is not necessary to destruct all params. We can group common parameters under ...rest and send them using only one variable.
 
`
 const {
    form: { handleSubmit },
    onClickNext,
    onSubmit,
    requireBasicCompanyDetails,
    thirdFile,
    forthFile,
    ...rest
  } = useStep1FormHook()

  return (
    <form className="row" onSubmit={handleSubmit(onSubmit)}>
      {/* TODO: Send details from the hook as props when continuing the flow without company information */}
      {requireBasicCompanyDetails && <BasicCompanyDetails />}
      <UploadFiles {...rest} />
      <UploadFiles
        {...{ rest, thirdFile, forthFile }}
      />
`

Comment (iuliudascalu) [reply src/components/Step1/Step1Form.tsx]: Done.

**Review by andreir-suciu — APPROVED** [review]

File: src/components/Step1/Step1Form.tsx [hunk 2]
```diff
@@ -6,21 +6,18 @@ import UploadFiles from './UploadFiles'
...
   const {
-    dispatch,
-    firstFile,
     form: { handleSubmit },
-    onChange,
     onClickNext,
     onSubmit,
     requireBasicCompanyDetails,
-    secondFile,
+    ...rest
```
Comment (mariuspop86) [inline src/components/Step1/Step1Form.tsx:13]: 👍

File: src/components/Step1/UploadFiles.tsx [hunk 3]
```diff
@@ -26,6 +37,22 @@ const UploadFiles = ({
           dispatch={dispatch}
           label={<SecondUploadLabel />}
         />
+        <Upload
```
Comment (mariuspop86) [inline src/components/Step1/UploadFiles.tsx:40]: Please refactor this as disscused.

File: src/hooks/useStep1FormHook.ts [hunk 4]
```diff
@@ -5,25 +5,42 @@ import useStepFormHook from './useStepFormHook'
...
+  FORMULAIRE_AUTO_CERTIFICATION,
+  JUSTIFICATIF_REVENU,
+} = DOC_TYPE_ID
 
 const initialState: IFileState[] = [
   {
     file: null,
     message: '',
-    personTypeId: '1', // TODO update when we have the correct type
+    personTypeId: PERSON_TYPE_ID,
```
Comment (mariuspop86) [inline src/hooks/useStep1FormHook.ts:21]: Please changed this as discussed

File: src/components/Step1/UploadFiles.tsx (hunk 3 above)
Comment (iuliudascalu) [reply src/components/Step1/UploadFiles.tsx]: Done.

File: src/hooks/useStep1FormHook.ts (hunk 4 above)
Comment (iuliudascalu) [reply src/hooks/useStep1FormHook.ts]: Done.

File: src/components/Step1/BasicCompanyDetails.tsx [hunk 5]
```diff
@@ -1,21 +1,16 @@
...
-  } = useStep1FormHook()
 
+const BasicCompanyDetails = ({
+  companyType,
+  control,
+  formState: { errors },
+  register,
+  textInputLabel,
+  updateCompanyType,
+}: any): ReactElement => {
```
Comment (mariuspop86) [inline src/components/Step1/BasicCompanyDetails.tsx:13]: Please set an interface here instead of `any`

File: src/components/Step1/Step1Form.tsx [hunk 6]
```diff
@@ -6,17 +6,19 @@ import UploadFiles from './UploadFiles'
...
     ...rest
   } = useStep1FormHook()
 
   return (
-    <form className="row" onSubmit={handleSubmit(onSubmit)}>
+    <form className="row" onSubmit={form.handleSubmit(onSubmit)}>
       {/* TODO: Send details from the hook as props when continuing the flow without company information */}
-      {requireBasicCompanyDetails && <BasicCompanyDetails />}
+      {requireBasicCompanyDetails && (
+        <BasicCompanyDetails {...rest} {...form} />
```
Comment (mariuspop86) [inline src/components/Step1/Step1Form.tsx:20]: <BasicCompanyDetails {...{...rest,...form}} />

File: src/components/Step1/Step1UploadLabels/Step1UploadLabel.tsx [hunk 7]
```diff
@@ -0,0 +1,20 @@
+import React from 'react'
+import FirstUploadLabel from './FirstUploadLabel'
```
Comment (mariuspop86) [inline src/components/Step1/Step1UploadLabels/Step1UploadLabel.tsx:2]: You should create an index file that exports all of this components, and then the import will be more simple
` import {FirstUploadLabel,SecondUploadLabel... } from './'`

File: src/components/Step1/UploadFiles.tsx [hunk 8]
```diff
@@ -1,58 +1,24 @@
...
-  secondFile,
-  thirdFile,
-  forthFile,
-}: any): ReactElement => {
-  const montant: string = useSelector(
-    (state: IStore) => state.folderReducer.folder.montant
-  )
+import Step1UploadLabel from './Step1UploadLabels/Step1UploadLabel'
 
+const UploadFiles = ({ onChange, dispatch, state }: any): ReactElement => {
```
Comment (mariuspop86) [inline src/components/Step1/UploadFiles.tsx:5]: Please add a more specific interface here

File: src/hooks/useStep1FormHook.ts [hunk 9]
```diff
@@ -4,48 +4,14 @@ import { useSelector } from 'react-redux'
...
-    retry: true,
-    showDelete: false,
-    typeId: JUSTIFICATIF_REVENU,
-  },
-]
+const { FORMULAIRE_AUTO_CERTIFICATION } = DOC_TYPE_ID
 
 const useStep1FormHook = () => {
   const intl = useIntl()
+  const initialState = documentsInitialState
```
Comment (mariuspop86) [inline src/hooks/useStep1FormHook.ts:14]: Please update also at line 37.

File: src/components/Step1/BasicCompanyDetails.tsx (hunk 5 above)
Comment (iuliudascalu) [reply src/components/Step1/BasicCompanyDetails.tsx]: Done

File: src/components/Step1/Step1Form.tsx (hunk 6 above)
Comment (iuliudascalu) [reply src/components/Step1/Step1Form.tsx]: Done

File: src/components/Step1/UploadFiles.tsx (hunk 8 above)
Comment (iuliudascalu) [reply src/components/Step1/UploadFiles.tsx]: Done

File: src/hooks/useStep1FormHook.ts (hunk 9 above)
Comment (iuliudascalu) [reply src/hooks/useStep1FormHook.ts:14]: Done

File: src/components/Step1/UploadFiles.tsx [hunk 10]
```diff
@@ -1,31 +1,29 @@
...
-          label={<FirstUploadLabel />}
-        />
-        <Upload
-          onChange={onChange}
-          fileState={secondFile}
-          index={1}
-          dispatch={dispatch}
-          label={<SecondUploadLabel />}
-        />
+        {state.map((item: any, index: number) => {
```
Comment (mariuspop86) [inline src/components/Step1/UploadFiles.tsx:14]:
```suggestion
        {state.map((item: IFileState, index: number) => {
```

File: src/interfaces/UploadFiles.ts [hunk 11]
```diff
@@ -0,0 +1,8 @@
+import { Dispatch } from 'react'
+import { Action, IFileState } from './Upload'
+
+export interface IUploadFiles {
```
Comment (mariuspop86) [inline src/interfaces/UploadFiles.ts:4]:
```suggestion
export interface IUploadFilesProps {
```

File: src/interfaces/BasicCompanyDetails.ts [hunk 12]
```diff
@@ -0,0 +1,10 @@
+import { Control, FieldErrors, UseFormRegister } from 'react-hook-form'
+
+export interface IBasicCompanyDetails {
```
Comment (mariuspop86) [inline src/interfaces/BasicCompanyDetails.ts:3]:
```suggestion
export interface IBasicCompanyDetailsProps {
```

**Review by mariuspop86 — APPROVED** [review]

**Pull request merged**